### PR TITLE
Week1 backend start

### PR DESCRIPTION
**Pull request by mchristia**
- Surf Spot Controller
- Surf Spot Service
- Sur Spot Object
- Test for Service adding a Surf Spot to Repository

### CODE REVIEW

**Review by christian-porsch — COMMENTED** [review]
LGTM 👍

File: backend/src/main/java/de/neuefische/backend/controller/SurfSpotController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public class SurfSpotController {
+
+    private final SurfSpotService surfSpotService;
+
+    @Autowired
+    public SurfSpotController(SurfSpotService surfSpotService){
+        this.surfSpotService = surfSpotService;
+    }
+
+    @GetMapping()
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/backend/controller/SurfSpotController.java]:
```suggestion
    @GetMapping
```

File: backend/src/main/java/de/neuefische/backend/controller/SurfSpotController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public List<SurfSpot> getAllSurfSpot(){
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/backend/controller/SurfSpotController.java]:
```suggestion
    public List<SurfSpot> getAllSurfSpots(){
```

File: backend/src/main/java/de/neuefische/backend/model/SurfSpot.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+public class SurfSpot {
+    private String id;
+    private String name;
+    private int longitude;
+    private int latitude;
+//    private Geography geography;
+//    private Weather weather;
+//    private Waves waves;
```
Comment (christophersiem) [inline backend/src/main/java/de/neuefische/backend/model/SurfSpot.java:17]:
```suggestion

```

File: backend/src/test/java/de/neuefische/backend/service/SurfSpotServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class SurfSpotServiceTest {
+
+    private SurfSpotRepo surfSpotRepo = new SurfSpotRepo();
+    private  SurfSpotService surfSpotService = new SurfSpotService(surfSpotRepo);
+
+    @Test
```
Comment (christophersiem) [inline backend/src/test/java/de/neuefische/backend/service/SurfSpotServiceTest.java:16]: missing test for adding a surfspot

File: backend/src/test/java/de/neuefische/backend/service/SurfSpotServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                new SurfSpot("GHI", "Spot1", 1, 3)
+        );
+
+        surfSpotRepo.addSurfSpot(surfSpotList.get(0));
+        surfSpotRepo.addSurfSpot(surfSpotList.get(1));
+        surfSpotRepo.addSurfSpot(surfSpotList.get(2));
+
+        List<SurfSpot> expectedList;
+        //When
+        expectedList = surfSpotService.getAllSurfSpots();
```
Comment (christophersiem) [inline backend/src/test/java/de/neuefische/backend/service/SurfSpotServiceTest.java]:
```suggestion
        //When
         List<SurfSpot> actualList = surfSpotService.getAllSurfSpots();
```

**Review by christophersiem — COMMENTED** [review]
looking good, added a few suggestions :)

File: backend/src/main/java/de/neuefische/backend/controller/SurfSpotController.java (hunk 1 above)
Comment (mchristia) [reply backend/src/main/java/de/neuefische/backend/controller/SurfSpotController.java]: thx

File: backend/src/main/java/de/neuefische/backend/controller/SurfSpotController.java (hunk 2 above)
Comment (mchristia) [reply backend/src/main/java/de/neuefische/backend/controller/SurfSpotController.java]: thx

File: backend/src/main/java/de/neuefische/backend/model/SurfSpot.java (hunk 3 above)
Comment (mchristia) [reply backend/src/main/java/de/neuefische/backend/model/SurfSpot.java:17]: wanted to keep this as reminder what i might need to add in this class

File: backend/src/test/java/de/neuefische/backend/service/SurfSpotServiceTest.java (hunk 4 above)
Comment (mchristia) [reply backend/src/test/java/de/neuefische/backend/service/SurfSpotServiceTest.java:16]: thx

File: backend/src/test/java/de/neuefische/backend/service/SurfSpotServiceTest.java (hunk 5 above)
Comment (mchristia) [reply backend/src/test/java/de/neuefische/backend/service/SurfSpotServiceTest.java]: thx

**Pull request merged**